ow_plexil/src/plexil-adapter/LanderAdapter.cpp

```diff
@@ -0,0 +1,274 @@
+}
+
+static void arm_stop (Command* cmd, AdapterExecInterface* intf)
+{
+  std::unique_ptr<CommandRecord>& cr = new_command_record(cmd, intf);
+  LanderAdapter::s_interface->armStop (CommandId);
+  acknowledge_command_sent(*cr);
+}
+
+static void arm_stow (Command* cmd, AdapterExecInterface* intf)
+{
+  std::unique_ptr<CommandRecord>& cr = new_command_record(cmd, intf);
+  LanderAdapter::s_interface->armStow (CommandId);
+  acknowledge_command_sent(*cr);
+}
+
+static void arm_move_cartesian (Command* cmd, AdapterExecInterface* intf)
+{
+  int frame;
+  bool relative;
+  vector<double> const *position_vector = nullptr;
+  vector<double> const *orientation_vector = nullptr;
+  RealArray const *position = nullptr;
+  RealArray const *orientation = nullptr;
+  // Get arguments.
+  const vector<Value>& args = cmd->getArgValues();
+  args[0].getValue(frame);
+  args[1].getValue(relative);
+  args[2].getValuePointer(position);
+  args[3].getValuePointer(orientation);
+  // Change real array into a vector.
+  position->getContentsVector(position_vector);
+  orientation->getContentsVector(orientation_vector);
+  std::unique_ptr<CommandRecord>& cr = new_command_record(cmd, intf);
+  LanderAdapter::s_interface->armMoveCartesian (frame, relative,
+                                                *position_vector,
+                                                *orientation_vector,
+                                                CommandId);
+  acknowledge_command_sent(*cr);
+}
+
+static void arm_move_cartesian_guarded (Command* cmd,
+                                        AdapterExecInterface* intf)
+{
+  int frame;
+  bool relative;
+  vector<double> const *position_vector = nullptr;
+  vector<double> const *orientation_vector = nullptr;
+  double force_threshold, torque_threshold;
+  RealArray const *position = nullptr;
+  RealArray const *orientation = nullptr;
+  const vector<Value>& args = cmd->getArgValues();
+  args[0].getValue(frame);
+  args[1].getValue(relative);
+  args[2].getValuePointer(position);
+  args[3].getValuePointer(orientation);
+  args[4].getValue(force_threshold);
+  args[5].getValue(torque_threshold);
+
+    //change real array into a vector
+  position->getContentsVector(position_vector);
+  orientation->getContentsVector(orientation_vector);
+  std::unique_ptr<CommandRecord>& cr = new_command_record(cmd, intf);
+  LanderAdapter::s_interface->armMoveCartesianGuarded (frame, relative,
+                                                       *position_vector,
+                                                       *orientation_vector,
+                                                       force_threshold,
+                                                       torque_threshold,
+                                                       CommandId);
+  acknowledge_command_sent(*cr);
+}
+
+static void arm_move_joint (Command* cmd, AdapterExecInterface* intf)
+{
+  bool relative;
+  int joint;
+  double angle; //radians
```

Comment:
Better to name it `angle_rad` or something that suggests to the programmer how it must be used.

Reply:
Good suggestion, done.